src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticAnalysisContext.cs

```diff
@@ -1166,11 +1171,17 @@ public struct OperationAnalysisContext
         /// </summary>
         public CancellationToken CancellationToken => _cancellationToken;
 
+        [Obsolete("Prefer the constructor that takes SemanticModel instead of Compilation")]
```

Comment:
>[Obsolete("Prefer the constructor that takes SemanticModel instead of Compilation")] [](start = 8, length = 84)

So, we are marking a public constructor as obsolete and suggesting to use a different constructor that is internal?

Reply:
@mavasani Do you know why these constructors are public?  it seems like that's not actually intentional (similar to how some compiler Enumerator structs had public constructors).

Reply:
When we had originally designed the analysis contexts, the constructors were kept public to enable custom analyzer drivers (for custom analyzer hosts) to execute analyzers. Let me try to track down the issue/discussion around this.

Reply:
Making the other ctor `public` is fine with me of course, the damage is already done if it should not have been public.

Reply:
We should not mark this as deprecated. In future, if we support `MSIL -> IOperation`, the analysis context has no semantic model, and must use this constructor. In fact this was the primary reasoning that John Hamby pushed back on exposing semantic model off any operation analysis contexts.